### ISSUE TITLE
Configure validation with a config file

### ISSUE THREAD

**Issue by jackdelahunt** [aerogear/charmil#141]
### Feature
Currently we building validation rules is very manual. Exporting this to a config file and then loading that data when we create the validator.
```
ruleCfg := rules.ValidatorConfig{
	ValidatorRules: rules.ValidatorRules{
		Length: rules.Length{Limits: map[string]rules.Limit{"Use": {Min: 1}}},
		MustExist: rules.MustExist{Fields: map[string]bool{"Args": true}},
		UseMatches: rules.UseMatches{Regexp: `^[^-_+]+$`},
	},
}
```

### Proposed solution
It would be nice to build this struct from a config file along the lines of  this example. To give users of charmil a higher level overview of the rules and make sharing and copying the rules easier.
```
{
         customRules: [
		{
			name: "mustExist",
			enabled: false,
			verbose: true,
			fields: [
				"Use",
				"Short",
				"Long",
				"Example",
			],
		},
		{
			name: "length",
			enabled: true,
			verbose: false,
			minLength: 5,
			maxLength: 10
		}
	]
}
```

### Additional context
Namit has a good example of loading json files [here](https://github.com/aerogear/charmil/pull/108/files?file-filters%5B%5D=.go&file-filters%5B%5D=.mod&file-filters%5B%5D=.sum&file-filters%5B%5D=.yaml&file-filters%5B%5D=dotfile#diff-d23cf55c813037e0e61bf2f04c179561077e5cef247ac4305ffc9dab37ed351dR97-R124)

**Comment by ankithans** [comment]
see the conversation here https://github.com/aerogear/charmil/pull/105
related?

**Comment by jackdelahunt** [comment]
Yes but I can only see parsing a string of json rather then a whole config file. But this is essentially the exact thing this issue is about.

**Comment by jackdelahunt** [comment]
> in golang struct do you want this?

All this issue is about is instead of parsing the json as we do from a string in the code. Load it from a file and parse that.

**Comment by ankithans** [comment]
> All this issue is about is instead of parsing the json as we do from a string in the code. Load it from a file and parse that.

this is for the end user API? We can give them 2 functions, one accepting the config from json/yaml, other accepting from golang struct(already there)

**Comment by jackdelahunt** [comment]
> this is for the end user API? We can give them 2 functions, one accepting the config from json/yaml, other accepting from golang struct(already there)

Yep pretty much. Instead of needing the construct the rule config from a struct you can use a config file instead.

**Comment by jackdelahunt** [comment]
Including Namit's pr we currently are loading resources from config files in two different places and with this it could be a third.
- [Localizer](https://github.com/aerogear/charmil/blob/main/core/localize/i18n.go#L52-L86)
- [Config loader](https://github.com/aerogear/charmil/pull/108/files?file-filters%5B%5D=.go&file-filters%5B%5D=.mod&file-filters%5B%5D=.sum&file-filters%5B%5D=.yaml&file-filters%5B%5D=dotfile#diff-d23cf55c813037e0e61bf2f04c179561077e5cef247ac4305ffc9dab37ed351dR97-R124)

Maybe extracting these out to reduce code duplication is also something that may need to be done.

**Comment by namit-chandwani** [comment]
> Maybe extracting these out to reduce code duplication is also something that may need to be done.

Exactly, that is the plan. 
Once my PR gets merged, I'll modify the other packages of core where config is involved and make them read/write from/to the single centrally maintained local config file

**Comment by wtrocki** [comment]
FYI

> Localizer
> Config loader

Fist one is developent config for cli, second is for end users of cli. Different purposes.

As for json config, loaded from file ours struct does provide metadata for parsing so anyone can do it right now etc. We might add yaml metadata to support it as well 


I'm proponent of strongly typed APIs that have one snd clear way to interact with logic. loading json from file or parsing string to struct can be done on user space

**Comment by wtrocki** [comment]
TLDR

> To give users of charmil a higher level overview of the rules and make sharing and copying the rules easier.

I think that is good idea that can be done in some helper while our api still accept struct

**Comment by jackdelahunt** [comment]
> I think that is good idea that can be done in some helper method while our api still accept struct

Cool will start this tomorrow then
